packages/core/src/agent/Dispatcher.ts

```diff
@@ -91,17 +92,21 @@ class Dispatcher {
   }
 
   private getHandlerForType(messageType: string): Handler | undefined {
+    const incomingMessageType = parseMessageType(messageType)
+
     for (const handler of this.handlers) {
       for (const MessageClass of handler.supportedMessages) {
-        if (MessageClass.type === messageType) return handler
+        if (canHandleMessageType(MessageClass, incomingMessageType)) return handler
       }
     }
   }
```

Comment:
I think this might be a big cleaner


```ts
return this.handlers.forEach((handler) => handler.supportedMessages.find((message) => message.type === messageType))
```

Reply:
This won't work because the forEach doesn't return anything. I could change it to find, but then it would return the handler and not the message class. 

Any other suggestions on how to approach?

Reply:
I believe that swapping the `forEach` with a find would work.